spot_driver/launch/spot_driver.launch.py

```diff
@@ -193,57 +184,7 @@ def generate_launch_description() -> launch.LaunchDescription:
             description="Choose whether to launch the image publishing nodes from Spot.",
         )
     )
-    launch_args.append(
-        DeclareLaunchArgument(
-            "depth_registered_mode",
-            default_value="from_nodelets",
-            choices=["disable", "from_spot", "from_nodelets"],
-            description=(
-                "If `disable` is set, do not publish registered depth images."
-                " If `from_spot` is set, request registered depth images from Spot through its SDK."
-                " If `from_nodelets` is set, use depth_image_proc::RegisterNode component nodes running on the host"
-                " computer to create registered depth images (this reduces the computational load on Spot's internal"
-                " systems)."
-            ),
-        )
-    )
-    launch_args.append(
-        DeclareLaunchArgument(
-            "publish_point_clouds",
-            default_value="False",
-            choices=["True", "true", "False", "false"],
-            description=(
-                "If true, create and publish point clouds for each depth registered and RGB camera pair. Requires that"
-                " the depth_register_mode launch argument is set to a value that is not `disable`."
-            ),
-        )
-    )
-    launch_args.append(
-        DeclareLaunchArgument(
-            "uncompress_images",
-            default_value="True",
-            choices=["True", "true", "False", "false"],
-            description="Choose whether to publish uncompressed images from Spot.",
-        )
-    )
-    launch_args.append(
-        DeclareLaunchArgument(
-            "publish_compressed_images",
-            default_value="False",
-            choices=["True", "true", "False", "false"],
-            description="Choose whether to publish compressed images from Spot.",
-        )
-    )
-    launch_args.append(
-        DeclareLaunchArgument(
-            "stitch_front_images",
-            default_value="False",
-            choices=["True", "true", "False", "false"],
-            description=(
-                "Choose whether to publish a stitched image constructed from Spot's front left and right cameras."
-            ),
-        )
-    )
+    launch_args += declare_image_publisher_args()
```

Comment:
I realize this is not the most ROS-y way to do this but I wasn't sure of a better way to reduce all of this repeated boilerplate

Reply:
This looks great to me. I wouldn't worry about the most "ROS-y" way and just focus on what is effective and maintainable.